src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -710,6 +710,20 @@ public function install_package( $args = array() ) {
 		if ( is_wp_error( $res ) ) {
 			$this->result = $res;
 			return $res;
+		} elseif ( $this instanceof Plugin_Upgrader && ! empty( $this->new_plugin_data['File'] ) ) {
+			$dependencies = false;
+			if ( ! empty( $this->new_plugin_data['RequiresPlugins'] ) ) {
+				$dependencies = $this->new_plugin_data['RequiresPlugins'];
+				$dependencies = WP_Plugin_Dependencies::sanitize_dependency_slugs( $dependencies );
+			}
+
+			update_site_option(
+				'wp_plugin_dependencies_known_dependencies',
+				array_merge(
+					get_site_option( 'wp_plugin_dependencies_known_dependencies', array() ),
+					array( $this->new_plugin_data['File'] => $dependencies )
```

Comment:
@afragen I _believe_ this is correct. See the `Plugin_Upgrader::check_package()` changes above for what the `File` key contains.

This should correspond to this in `WP_Plugin_Dependencies::read_dependencies_from_plugin_headers()`:

```php
foreach ( array_keys( self::$plugins ) as $plugin ) {
```

Where `self::$plugins` is the result of `get_plugins()`, which should contain `plugin-dir/plugin_file.php`.